captum/optim/_param/image/transform.py

```diff
@@ -459,3 +464,76 @@ def __init__(self, warp: bool = False) -> None:
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         assert x.dim() == 4
         return nchannels_to_rgb(x, self.warp)
+
+
+class RandomCrop(nn.Module):
+    """
+        Randomly crop out a specific size from an NCHW image tensor.
+    ​
+        Args:
+            crop_size (int, sequence, int): The desired cropped output size.
+    """
+
+    def __init__(
+        self,
+        crop_size: Union[Tuple[int, int], Tuple[int]],
+    ) -> None:
+        super().__init__()
+        self.crop_size = crop_size
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        assert x.dim() == 4
+        hs = x.shape[2] - self.crop_size[0]
+        ws = x.shape[3] - self.crop_size[1]
+        shifts = [
+            torch.randint(low=-hs, high=hs, size=[1]),
+            torch.randint(low=-ws, high=ws, size=[1]),
+        ]
+        x = torch.roll(x, shifts, dims=(2, 3))
+        return center_crop(
+            x,
+            crop_vals=self.crop_size,
+            pixels_from_edges=False,
+        )
+
+
+class AlphaChannelLoss(nn.Module):
+    """
+        Transform for calculating alpha channel loss, while passing the input tensor
+        through unaltered. Loss values are calculated in such a way that opague and
+        transparent regions of the tensor are automatically balanced.
+    ​
+        See: https://distill.pub/2018/differentiable-parameterizations/
+        Mordvintsev, et al., "Differentiable Image Parameterizations", Distill, 2018.
+    ​
+        Args:
+            scale (float, sequence): Tuple of rescaling values to randomly select from.
+            crop_size (int, sequence, int, optional): The desired cropped output size
+                for secondary alpha channel loss.
+            background (tensor, optional):  An NCHW image tensor to be used as the
+                alpha channel's background.
+    """
+
+    def __init__(
+        self,
+        scale: NumSeqOrTensorType,
+        crop_size: Optional[Tuple[int, int]] = None,
+        background: Optional[torch.Tensor] = None,
+    ) -> None:
+        super().__init__()
+        self.random_scale = RandomScale(scale=scale)
+        self.crop_size = crop_size
+        self.random_crop = RandomCrop(crop_size)
+        self.blend_alpha = BlendAlpha(background=background)
+        self.loss = 0
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
```

Comment:
We should have two asserts here to make sure that the input has the correct shape:
 
```
assert x.dim() == 4
assert x.size(1) == 4
```

Reply:
Added assert checks with comments